pkgs/desktops/deepin/dtkcore/default.nix

```diff
@@ -3,41 +3,39 @@
 stdenv.mkDerivation rec {
   name = "${pname}-${version}";
   pname = "dtkcore";
-  version = "2.0.10";
+  version = "2.0.12.1";
 
   src = fetchFromGitHub {
     owner = "linuxdeepin";
     repo = pname;
     rev = version;
-    sha256 = "0dwpq6c38gaa95mgjnwj3vjz57n0cz6jfk950xi6s9ww2f4g6kq7";
+    sha256 = "1akfzkdhgsndm6rlr7snhpznxj0w351v6rr8vvnr6ka2dw75xsl4";
   };
 
   nativeBuildInputs = [
     pkgconfig
     qmake
     pythonPackages.wrapPython
+    deepin.setupHook
   ];
 
   buildInputs = [
     gsettings-qt
   ];
 
   postPatch = ''
-    # Only define QT_HOST_DATA if it is empty
-    sed '/QT_HOST_DATA=/a }' -i src/dtk_module.prf
-    sed '/QT_HOST_DATA=/i isEmpty(QT_HOST_DATA) {' -i src/dtk_module.prf
+    searchHardCodedPaths  # debugging
```

Comment:
this should stay around? last time I saw it, it garbled build output…

Reply:
`searchHardCodedPaths` should probably be adapted so that it doesn't touch binaries.

Reply:
> this should stay around? last time I saw it, it garbled build output…

Deepin makes heavy use of hard coded FHS paths in its code (like `/usr/lib/dde-daemon/`). This shell function help to locate them. The build output is not exactly garbled. It is just extended with information of possible occurrences of them. The use of colors in the output of grep may make it seem garbled when seen in an environment that does not show ASCII colors. But it has been very valuable for development, and re-adding it every time is not viable. Therefore I prefer to keep them.

Reply:
I think it's a good idea too, but maybe it's distracting when it's against binary's

For example with `dtkcore`
```
----------- looking for command invocations
----------- looking for hard coded paths
/nix/store/px60s5qx2sb78p0333qr7cn8hnahwl6r-dtkcore-2.0.12.1/lib/libdtkcore.so.2.0.6:u�H�H�1H�H;��tHH;Y�tWH;P�u�H�u����[���H�QH�����H�QH������H�u���f.�H�u���f�USH�H��t=H��H�5C�H��H���������u
                              H�H��[]�f�H�H��H��[]�����1���@ATUI��SH�͉�H��������x#��u+����H��1�L��D$
                   ������D$
                           ��H��[]A\���
                                       u���H�U�������H�H��/etc/deepin-versionVersionTypeEditionCopyrightDesktopProfessionalPRETTY_NAME=VERSION_ID=/etc/os-release/usr/lib/os-releaseDISTRIB_DESCRIPTION=DISTRIB_RELEASE=DISTRIB_ID=/etc/lsb-releasedeepindebianarchcentosfedoralinuxmintmanjaroopensusesailfishosubuntu/proc/cpuinfomodel name-Jlpb-oNAME,KNAME,PKNAME,SIZElsblkblockdevicespknamesizeX���������������0��������������������p����������������������P�����������������������������������X���������������0��������������������p����������������������P��N3Dtk4Core7DObjectEN3Dtk4Core14DObjectPrivateEN3Dtk4Core22DAbstractUnitFormatterEN3Dtk4Core18DTimeUnitFormatterE����d����h����m����s�?�-���q=�C��l��m��m���l���l������TB����GB����MB����KB����BN3Dtk4Core18DDiskSizeFormatterESt11_Mutex_baseILN9__gnu_cxx12_Lock_policyE2EESt16_Sp_counted_baseILN9__gnu_cxx12_Lock_policyE2EESt19_Sp_make_shared_tagSt23_
```

Reply:
Added an option (`-a`) to enable looking for hints of FHS hard coded files in binary files in helper function `searchHardCodedPaths`. By default it does no look at them binary files.